Source/Core/Common/CompatPatches.cpp

```diff
@@ -162,37 +164,26 @@ struct Version
   }
 };
 
-static bool GetModulePath(const wchar_t* name, std::wstring* path)
+static std::optional<std::wstring> GetModulePath(const wchar_t* name)
 {
   auto module = GetModuleHandleW(name);
   if (module == nullptr)
-    return false;
-  DWORD path_len = MAX_PATH;
-retry:
-  path->resize(path_len);
-  path_len = GetModuleFileNameW(module, const_cast<wchar_t*>(path->data()),
-                                static_cast<DWORD>(path->size()));
-  if (!path_len)
-    return false;
-  auto error = GetLastError();
-  if (error == ERROR_SUCCESS)
-    return true;
-  if (error == ERROR_INSUFFICIENT_BUFFER)
-    goto retry;
-  return false;
+    return {};
```

Comment:
```suggestion
    return std::nullopt;
```

Default initialization and the constructor taking std::nullopt_t are [not](https://godbolt.org/z/vOIZmE) behaviorally equivalent

Reply:
This page seems to say otherwise
https://en.cppreference.com/w/cpp/utility/optional/optional
as both `optional()` and `optional( std::nullopt_t )` are linked to the same description.

Reply:
They provide the same basic observable behavior, but the reality of the situation is that they currently, absolutely, do not behave the same way in how they accomplish that visible behavior (*especially* with [large](https://godbolt.org/z/tBkOGO) objects)

Please use std::nullopt in this case.

Reply:
I will, I was just curious about real life examples like the one you provided. This is a very good example towards it indeed.